src/app/services/baw-api/harvest/harvest.service.ts

```diff
@@ -0,0 +1,194 @@
+   *
+   * @param status Status to change to if valid
+   * @param project Project of harvest
+   */
+  public transitionStatus(
+    status: HarvestStatus,
+    project: IdOr<Project>
+  ): Observable<Harvest> {
+    return this.api
+      .httpPatch(
+        endpoint(project, emptyParam, emptyParam),
+        // TODO Replace harvest with Harvest.kind
+        { harvest: { status } }
+      )
+      .pipe(map(this.api.handleSingleResponse(Harvest)));
+  }
+
+  /**
+   * Update the mappings of a harvest
+   *
+   * @param mappings Mappings of folders to sites
+   * @param project Project of harvest
+   */
+  public updateMappings(
+    mappings: IHarvestMapping[],
+    project: IdOr<Project>
+  ): Observable<Harvest> {
+    return this.api
+      .httpPatch(
+        endpoint(project, emptyParam, emptyParam),
+        // TODO Replace harvest with Harvest.kind
+        { harvest: { mappings } }
+      )
+      .pipe(map(this.api.handleSingleResponse(Harvest)));
+  }
+}
+
+@Injectable()
+export class ShallowHarvestsService implements StandardApi<Harvest> {
+  public constructor(private api: BawApiService<Harvest>) {}
+
+  public list(): Observable<Harvest[]> {
+    return this.api.list(Harvest, shallowEndpoint(emptyParam, emptyParam));
+  }
+
+  public filter(filters: Filters<Harvest>): Observable<Harvest[]> {
+    return this.api.filter(
+      Harvest,
+      shallowEndpoint(emptyParam, filterParam),
+      filters
+    );
+  }
+
+  public show(model: IdOr<Harvest>): Observable<Harvest> {
+    return this.api.show(Harvest, shallowEndpoint(model, emptyParam));
+  }
+
+  public create(model: Harvest): Observable<Harvest> {
+    return this.api.create(
+      Harvest,
+      shallowEndpoint(emptyParam, emptyParam),
+      (harvest) => shallowEndpoint(harvest, emptyParam),
+      model
+    );
+  }
+
+  public update(model: Harvest): Observable<Harvest> {
+    return this.api.update(Harvest, shallowEndpoint(model, emptyParam), model);
+  }
+
+  public destroy(model: IdOr<Harvest>): Observable<Harvest | void> {
+    return this.api.destroy(shallowEndpoint(model, emptyParam));
+  }
+
+  /**
+   * Transition the status of a harvest to the next step
+   *
+   * @param status Status to change to if valid
+   */
+  public transitionStatus(status: HarvestStatus): Observable<Harvest> {
+    return this.api
+      .httpPatch(
+        shallowEndpoint(emptyParam, emptyParam),
+        // TODO Replace harvest with Harvest.kind
+        { harvest: { status } }
+      )
+      .pipe(map(this.api.handleSingleResponse(Harvest)));
+  }
+
+  /**
+   * Update the mappings of a harvest
+   *
+   * @param mappings Mappings of folders to sites
+   */
+  public updateMappings(mappings: IHarvestMapping[]): Observable<Harvest> {
+    return this.api
+      .httpPatch(
+        shallowEndpoint(emptyParam, emptyParam),
+        // TODO Replace harvest with Harvest.kind
+        { harvest: { mappings } }
+      )
+      .pipe(map(this.api.handleSingleResponse(Harvest)));
```

Comment:
this duplication is messy. I'd say either find a way to factor out the commonality or only define these methods on one of the services. Shallow is probably a good choice because both only apply to an instance of  a harvest.